reth-rpc won't build due to renamed enum in reth-primitives

### Describe the bug

`revm_primitives::InvalidTransaction::GasMaxFeeGreaterThanPriorityFee` was renamed to `PriorityFeeGreaterThanMaxFee` [upstream](https://github.com/bluealloy/revm/pull/722). This breaks `reth-rpc` which pattern matches this variant. The build breakage is currently masked by `Cargo.lock`, but that doesn't help people trying to build on top of Reth.

### Steps to reproduce

1. Checkout Reth
2. `rm Cargo.lock`
3. `cargo build`

```
   Compiling reth-rpc v0.1.0-alpha.8 (/home/ptd/Projects/reth/crates/rpc/rpc)
error[E0599]: no variant or associated item named `GasMaxFeeGreaterThanPriorityFee` found for enum `revm_primitives::InvalidTransaction` in the current scope
   --> crates/rpc/rpc/src/eth/error.rs:373:33
    |
373 |             InvalidTransaction::GasMaxFeeGreaterThanPriorityFee => {
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `InvalidTransaction`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `reth-rpc` (lib) due to previous error
```

### Node logs

_No response_

### Platform(s)

Linux (x86)

### What version/commit are you on?

f00df57ea5f5c1a1957be7498fabc9442f26c8d7

### What database version are you on?

_No response_

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

--- a/crates/rpc/rpc/src/eth/error.rs
+++ b/crates/rpc/rpc/src/eth/error.rs
@@ -370,7 +370,7 @@ impl From<revm::primitives::InvalidTransaction> for RpcInvalidTransactionError {
         use revm::primitives::InvalidTransaction;
         match err {
             InvalidTransaction::InvalidChainId => RpcInvalidTransactionError::InvalidChainId,
-            InvalidTransaction::GasMaxFeeGreaterThanPriorityFee => {
+            InvalidTransaction::PriorityFeeGreaterThanMaxFee => {
                 RpcInvalidTransactionError::TipAboveFeeCap
             }
             InvalidTransaction::GasPriceLessThanBasefee => RpcInvalidTransactionError::FeeCapTooLow,

